lang/fpc/files/t_darwin.pas.patch

```diff
@@ -0,0 +1,32 @@
+--- compiler/systems/t_darwin.pas.orig	2021-04-10 13:02:51
++++ compiler/systems/t_darwin.pas	2023-10-10 11:43:18
+@@ -123,16 +123,16 @@
+              programs with problems that require Valgrind will have more
+              than 60KB of data (first 4KB of address space is always invalid)
+            }
+-           ExeCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $DYNLINK $STATIC $GCSECTIONS $STRIP $MAP $ORDERSYMS -multiply_defined suppress -L. -o $EXE $CATRES $FILELIST';
++           ExeCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $DYNLINK $STATIC $GCSECTIONS $STRIP $MAP $ORDERSYMS -L. -o $EXE $CATRES $FILELIST';
+            if not(cs_gdb_valgrind in current_settings.globalswitches) then
+              ExeCmd[1]:=ExeCmd[1]+' -pagezero_size 0x10000';
+   {$else ndef cpu64bitaddr}
+-           ExeCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $DYNLINK $STATIC $GCSECTIONS $STRIP $MAP $ORDERSYMS -multiply_defined suppress -L. -o $EXE $CATRES $FILELIST';
++           ExeCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $DYNLINK $STATIC $GCSECTIONS $STRIP $MAP $ORDERSYMS -L. -o $EXE $CATRES $FILELIST';
+   {$endif ndef cpu64bitaddr}
+            if (apptype<>app_bundle) then
+-             DllCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $GCSECTIONS $MAP $ORDERSYMS -dynamic -dylib -multiply_defined suppress -L. -o $EXE $CATRES $FILELIST'
++             DllCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $GCSECTIONS $MAP $ORDERSYMS -dynamic -dylib -L. -o $EXE $CATRES $FILELIST'
+            else
+-             DllCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $GCSECTIONS $MAP $ORDERSYMS -dynamic -bundle -multiply_defined suppress -L. -o $EXE $CATRES $FILELIST';
++             DllCmd[1]:='ld $PRTOBJ $TARGET $EMUL $OPT $GCSECTIONS $MAP $ORDERSYMS -dynamic -bundle -L. -o $EXE $CATRES $FILELIST';
+            DllCmd[2]:='strip -x $EXE';
+            DynamicLinker:='';
+          end;
+@@ -360,7 +360,7 @@
+       end;
+       if MacOSXVersionMin<>'' then
+         begin
+-          LinkRes.Add('-macosx_version_min');
++          LinkRes.Add('-macos_version_min');
```

Comment:
This one `fpc`-specific, right? Usually the flag is `-mmacosx-version-min=`, AFAIK.

Reply:
These are flags sent to the linker, and they are not fpc-specific.

From man ld:
```
-macos_version_min version
             This is set to indicate the oldest macOS version that that the output is to be used on.  Specifying a later version enables the linker
             to assumes features of that OS in the output file.  The format of version is a macOS version number such as 10.9 or 10.14
```
and NB, on Sonoma:
```
 % ld -macosx_version_min
-macosx_version_min has been renamed to -macos_version_min
```

However, here you have a problem, though. 

If you only use the new name, older ld versions won't work. If you only use the old name, you will get warnings from the linker all the time, which may cause link errors, and eventually the old name won't be accepted at all.

There will probably have to be different flags sent, depending on how new the linker is.

Reply:
@kencu Thank you for clarifying.

@kamischi We need this change to be conditional, otherwise many systems will be broken.

Reply:
> @kencu Thank you for clarifying.

Always happy to be helpful.

Someone might do a project to see where the cutoff is for the linker to accept 

```
-macosx_version_min
```

without complaining about it.

Reply:
The patch is restricted to ${macos_version_major} >= 13}, i.e. Ventura and I was assuming that this implicitly catches the corresponding version of the linker. I admit that I did not check this assumption thoroughly.

Reply:
@kamischi That should be fine then, I guess.

Could we get rid of 10.9 target forced for everything below macOS 11?

Reply:
It does so for all systems <10.7. As much as I can can remember, it is ok for 10.7 - 10.11.

Reply:
@kamischi I refer to this portion now: https://github.com/kamischi/macports-ports/blob/aa303114d11ba4852dcd5f1db13b8519952d70ec/lang/fpc/Portfile#L264-L268

Reply:
Sorry. Got it now. Do you agree to the next commit?